src/bin/main.rs

```diff
@@ -328,27 +370,45 @@ fn main() {
                 fs::create_dir_all(file_path.parent().unwrap()).unwrap();
                 match fs::copy(&doc_file, &file_path) {
                     Ok(_) => {}
-                    Err(err) => eprintln!("  --> Error copying doc file {:?}: {}", doc_file, err),
+                    Err(err) => eprintln!(
+                        "ERROR: --> Problem copying doc file {:?}: {}",
+                        doc_file, err
+                    ),
                 }
             }
         }
     }
 }
 
+fn modify_path(path: &PathBuf, replace: &str) -> PathBuf {
+    let mut new_path = PathBuf::new();
+    let repl_parts: Vec<_> = replace.split('/').filter(|&x| !x.is_empty()).collect();
```

Comment:
Did you try create a `Path` instance using `Path::new` and iterate its components instead of splitting manually here?

Reply:
I think you also do not need to collect here, e.g.
```rust
let mut path = path.components();
let mut replace = Path::new(replace);
for (path, replace) in path.zip(replace) { ... }
for path in path { new_path.push(push); }
```

Reply:
I also wonder whether a fast path for returning `path` unmodified would be a sensible addition?

Reply:
Fast returning is good.

Regarding `Path` instead of splitting, it would probably not work due to dots being lost? I think I will have to try...

Reply:
Don't get it working using `Path`.

Reply:
Because dot has a special meaning for paths. I would probably suggest avoiding dot and asterisk then due to the ambiguity with the current directory and glob patterns. Maybe two other sigils could perform the same task?

Reply:
(Nit `return path.clone();` would probably better convey the intent compared to `return path.to_owend();` as `PathBuf` is already owning. But of course `impl ToOwned for T where T: Clone`...)

Reply:
> Because dot has a special meaning for paths.

That's what I mean with "not special", as it keeps it the normal meaning in paths. 😄 
I.e. if you leave out the dot part from the code, the result would be the same (except for `eprintln!`). Replacing the path component by a dot results in omitting the component.

Reply:
The advantage I see for '*' is, that it is not a legal characher in paths. E.g. '_' would collide.

Reply:
> Replacing the path component by a dot results in omitting the component.

While I can follow how to arrive at this interpretation, I find it much too clever as it is applied here. I think the design rationale for something that even a heavy user of Yarner will interact with maybe twice in a blue moon should be as simple and straight forward as possible. If using something complex but well know like regular expression is deemed inappropriate, going for the simple version that just lists all targets and sources (three in the case we are interested in) seems preferable IMHO.

But then again, we already presented these positions and hence let's leave it as it is.

> Personally, I see no problem in splitting "manually", so what is your reason to prefer the Path?

Using the existing primitive to interpret text as a path relays intent and makes sure to handle any special case, like platform-specific separators instead of slash.

> The advantage I see for '*' is, that it is not a legal characher in paths.

At least for POSIX, everything but / and \0 are legal characters in paths. Even more restricted versions generally restrict to valid UTF-8 or ASCII which still allows *. It is a glob pattern and has special meaning when interacting with a shell, but that does not prevent from being part of a path, i.e. "*" is also a valid file and directory name.

Reply:
For `Path`, dot etc. are not a problem it seems. It's just that I am not familiar with the `Components`, e.g. how to get their count, but I will just have to dig into the documentation deeper.

Reply:
> e.g. how to get their count, but I will just have to dig into the documentation deeper.

While not efficient, `Components` is an iterator and hence `Iterator::count` can be used. But if you `.collect()` into `Vec<Component>`, this should not be an issue as you can just use `.len()`.

Reply:
(The two-loops-zip version should also not need the length, should it?)

Reply:
Think I got it with `Path`. So I will go for underscore instead of asterisk, although it would work too.

Reply:
Ok, you were right. is works with a single dot, but not with more than one.

Reply:
You could still avoid collecting the components into a vector using the variant
```rust
let mut path = path.components();
let mut replace = Path::new(replace);
for (path, replace) in path.zip(replace.components().map(|c| c.as_os_str())) { ... }
for path in path { new_path.push(push); }
```
which also avoids the index checks.